final_task/rss_parser/Output_functions.py

```diff
@@ -0,0 +1,332 @@
+        "main source": item.main_source,
+    }
+    return json.dumps(news_dict)
+
+
+def getting_info_into_file(item):
+    """
+    Preparing information to be written into the file in more readable way
+    :param item:
+    :return:
+    """
+    novelty = "\n{0}.\nTitle: {1}\nDate: {2}\nLink: {3}\nDescription:\n {4}\nImages links:{5}\nAlternative text:{6}\n" \
+              "Main source: {7}" \
+        .format(item.number_of_novelty,
+                pprint.pformat(item.title_of_novelty, width=115),
+                item.time_of_novelty,
+                pprint.pformat(item.source_link, width=115),
+                pprint.pformat(item.description, width=115),
+                pprint.pformat(item.images_links),
+                pprint.pformat(item.alt_text, width=115),
+                item.main_source)
+    return novelty
+
+
+def getting_corrected_time(item):
+    """
+    Getting time in view %Y%m%d
+    :param item:
+    :return:
+    """
+    corrected_date = datetime.strptime(item.get('published', ''), '%a, %d %b %Y %X %z')
+    return corrected_date.strftime('%Y%m%d')
+
+
+def getting_time_for_json(item):
+    corrected_date = datetime.strptime(item, '%a, %d %b %Y %X %z')
+    return corrected_date.strftime('%Y%m%d')
+
+
+def reading_file(name_of_file):
+    with open(name_of_file, 'r', encoding='utf-8') as news_cache:
+        return news_cache.read()
+
+
+def writing_to_file(pack_of_news, filename):
+    """
+    Writing information into 2 files: News_cache.txt and News_cache_json.txt
+    Creating 2 files because it's easier to read information into computer from JSON file than another file
+    1. Open file
+    2. Check if file is empty or not
+    3. If empty - append whole information
+       If Not empty:
+              1. Check if the novelty in file
+              2. If in file: continue
+                 If Not in the file: Find out length of file (amount of news), put that number to incoming novelty
+    If that novelty exists it will go to another novelty in limit
+    It means that if you enter --limit 15 and 10 news are already in list it will add only 5 news to list
+    :param pack_of_news:
+    :param filename:
+    :return:
+    """
+    with open(filename, 'a', encoding='utf-8') as news_cache:
+        content = reading_file('News_cache.txt')
+        if not content:
+            print("CONTENT")
+            for item in pack_of_news:
+                with open('News_cache_json.txt', 'a', encoding='utf-8') as news_cache_json:
+                    news_cache_json.write(converting_novelty_to_json(item))
+                    news_cache_json.write('\n')
+                news_cache.write(getting_info_into_file(item))
+                news_cache.write("\n_ _ _")
+        else:
+            for item in pack_of_news:
+                if item.source_link in content:
+                    continue
+                else:
+                    length = sum(1 for line in open('News_cache_json.txt', 'r')) + 1
```

Comment:
If you are actually using JSON format to store news, then it is better to store it in `file.json`, not `.txt`

Reply:
Fixed, thanks.